presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -962,7 +981,11 @@ public void finishDelete(Session session, TableHandle tableHandle, Collection<Sl
 
         Optional<QualifiedObjectName> objectName = prefix.asQualifiedObjectName();
         if (objectName.isPresent()) {
-            return getView(session, objectName.get())
+            // redirectTable assumes a non-empty table name
+            if (objectName.get().getCatalogName().isEmpty() || objectName.get().getSchemaName().isEmpty() || objectName.get().getObjectName().isEmpty()) {
```

Comment:
This shouldn't be needed, since `asQualifiedObjectName()` only returns a present optional if the name is fully qualified.

Reply:
The comment `// redirectTable assumes a non-empty table name` is a little bit misleading. Here's actually to handle empty strings. Example query:
```SQL
SELECT * FROM hive.information_schema.views WHERE table_name = '';
```
Let me change the comment to
```
// View cannot exist
```
to be consistent with the one in `MetadataManager::getView`.